packages/react/src/button/Button.tsx

```diff
@@ -70,9 +73,23 @@ export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
         </Text>
       );
 
-    const presetProps = presets[preset];
+    const presetProps = appearances[appearance];
     const finalColorScheme = colorScheme ?? presetProps.colorScheme;
     const finalVariant = variant ?? presetProps.variant;
+    const content = icon ? (
+      <Box className={styles.icon({ size: size })}>{icon}</Box>
+    ) : (
```

Comment:
I don't like this API. It's the evil branching creeping into the component.

Now `<Button>` has 3 extension point `leftSection`, `rightSection` and `icon`. And they can't all work together... It now has if-you-use-this-you-can-use-that issue. I'd much rather prefer `<IconButton>` component or something else that clearly separates it from the `<Button>`.

Just my 2 cents...

cc: @radnan

Reply:
I don’t think this branching is necessary 

The purpose of the `icon` prop is to simply set the shape of the button - you can call it `square` in your head if that helps

But it has no relation to what the actual contents of the buttons are

Reply:
The `icon` prop is only meant to modify the padding of the button similar to how `size` does

Reply:
So it's a `boolean`? Still I'd need to manually pass it when I'm passing an icon as `children` right? And also remember to set `aria-label`?

Reply:
This is why I think having a single `icon` prop would be better for passing the icon rather than separate leftSection/rightSection props

And then we can make `aria-label` required if a ReactElement is passed as `icon` but no `children` is given.

Reply:
I addressed the `icon` and `iconPosition` part. How to address the `aria-label` part bhai?